### PR TITLE
docs: Add sidebar-navigable sub-sections to Event Reference section of API Reference documentation

### LINKED ISSUE #460: docs: redesign event reference
### Summary

seperate event reference into multiple types and sections

### What is the feature request for?

The documentation

### The Problem

There's too many events in a disorganized disarray to find a specific one very quicky.

### The Ideal Solution

There's a lot of events and events are important to disnake. Each type of event should have a section label and be listed in the sidebar.

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ChristopherJHart**
## Summary
Fix #460 by breaking up Event Reference section of documentation into sidebar-navigable sub-sections.

The new sub-sections are as follows:

* **Client Events** - Events related to the disnake client, like connect/disconnect/ready/resume/socket events.
* **Chat Message Events** - Events related to a message. New messages, typing events, edited/deleted messages, reaction events, etc.
* **Interaction Events** - Events related to application commands or other interactions.
* **Channel/Thread Events** - Events related to channel/thread CRUD events, as well as pins.
* **Guild Events** - Events related to stuff that happens in a guild.
* **Voice Events** - Events related to voice state updates.

There were a large number of events in the Guild Events section, so guild events are further broken up into the following sub-sections for organizational purposes:

* **Guild Generic Events** - Generic events that don't really fit elsewhere.
* **Guild User/Member Events** - Events related to a user/member doing something (joining/leaving the guild, updating presence/activity, ban/unban, etc.)
* **Guild Invite Events** - Events related to an invite for the guild.
* **Guild Settings Events** - Events to changes in settings to the guild, such as role CRUD, scheduled event CRUD, webhook CRUD, etc.
* **Guild Integration Events** - Events related to integration CRUD in the guild.

The guild sub-sections are ordered from top to bottom as shown above based on my opinion of what the *average* disnake user is likely interested in. I'm more than happy to reorder these sub-sections (or remove them entirely) if requested!

## Checklist
- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by onerandomusername — CHANGES_REQUESTED** [review]
Hello and thank you for contributing, its really quite appreciated!

I have a small comment right now about some of the grouping of events:
- webhook events should be on channels
- scheduled events maybe should be in their own guild subcategory
- I'm not entirely sure about the grouping of guild settings right now, @shiftinv

Again, thank you for this much needed contribution!

**Comment by ChristopherJHart** [comment]
Marking as WIP until all thoughts/questions/concerns in #460 are addressed.

**Comment by onerandomusername** [comment]
We no longer have to use `WIP` tags, as we can convert the pull request to a draft which prevents merge automatically

**Comment by ChristopherJHart** [comment]
Changes with the latest commits:

* `on_webhooks_update` moved to `Channels/Threads` because that makes logical sense.
* `on_guild_scheduled_event*` moved to a new section `Scheduled Events`.
* Events organized within each section alphabetically, then by add/edit/remove/bulk/raw method.

Short of any other concerns/feedback, I think we're just waiting on @shiftinv's thoughts regarding the Guild Settings section. I'll keep this in draft status until they respond with their thoughts!

**Comment by onerandomusername** [comment]
There's still a bit of ABCD ordering missing with the sections overall: 
image

Ideally, ordering the current categories would be like the following
- Client
- Channels/Threads (not decided on the naming, letting shiftinv for that)
- Interactions
- Guilds
  - Generics
  - Integrations
  - Invites (not sure if this should maybe be in the channel section)
  - Members/Users (I would reorder these as most terminology uses member for users that are in guilds)
  - Scheduled Events
  - Settings
- Messages
- Voice


I still feel like there's a bit more that could be done in these categories for visibility of events... but I'll have to sleep on that.

**Comment by shiftinv** [comment]
Thanks for tackling this! A few comments:
1. The `Client` category sounds a bit ambiguous, something like `Connection` would be better
1. Really can't decide about `Channels/Threads` either -- `Channels, Threads` (`Channels + Threads`?) might work
1. I feel like `Generics` could be called `General` instead (or... something like that). `Generics` mainly makes me think of `typing.Generic` :p
1. `on_user_update` is not a guild event (only fires for the bot user), once that's moved `Users/Members` can just be called `Members`
1. `on_voice_state_update` would be considered a guild event as well
1. `on_group_join` + `on_group_remove` don't exist anymore but were never removed from the documentation, this might be a good opportunity for that
1. @onerandomusername https://github.com/DisnakeDev/disnake/pull/636#issuecomment-1188621519 is `Client` appearing before `Channels` intentional, or should we sort things in a fully alphabetical order?
    Also, the invite events are fine for now, imo.
1. @onerandomusername https://github.com/DisnakeDev/disnake/pull/636#pullrequestreview-1042168047 the `Settings` section looks fine to me, anything in particular you're looking to change?

**Comment by onerandomusername** [comment]
> * @onerandomusername ([comment](https://github.com/DisnakeDev/disnake/pull/636#issuecomment-1188621519)) is `Client` appearing before `Channels` intentional, or should we sort things in a fully alphabetical order?
>   Also, the invite events are fine for now, imo.
> 

Intially an oversight, but it doeesn't feel wrong for Client to be above, to be quite honest.

> * @onerandomusername ([comment](https://github.com/DisnakeDev/disnake/pull/636#pullrequestreview-1042168047)) the `Settings` section looks fine to me, anything in particular you're looking to change?


The settings section has a bit too much in it for me IMO. For example, Settings currently contains roles, emojis, stickers, and stage_instances. I think these might be be better suited as the following categories for Guilds:

 - General
 - Emojis
 - Integrations
 - Invites
 - Members
 - Scheduled Events
 - Stage Instances
 - Stickers
 - Roles

**Comment by ChristopherJHart** [comment]
I *believe* I tackled everything covered thus far in the latest set of commits! Let me know if you'd like to see any further changes.

**Comment by onerandomusername** [comment]
There is a slight bit of alphabetization missing, aside from Client and Channel:
Current:
 
image

Goal:
- Client
- Channels/Threads
- Guilds
- Interactions
- Messages


@shiftinv do you think that Channels/Threads should go before Client?

**Comment by shiftinv** [comment]
> @shiftinv do you think that Channels/Threads should go before Client?

The structure you proposed looks fine, keeping the general client events on top is definitely better than starting with channel events.

File: docs/api.rst [hunk 1]
```diff
@@ -357,1065 +362,1111 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
 
-.. function:: on_guild_channel_update(before, after)
+    .. versionadded:: 2.0
 
-    Called whenever a guild channel is updated. e.g. changed name, topic, permissions.
+    :param payload: The raw event payload data.
+    :type payload: :class:`RawIntegrationDeleteEvent`
 
-    This requires :attr:`Intents.guilds` to be enabled.
+Interactions
```
Comment (shiftinv) [inline docs/api.rst:783]: I feel like `Application Commands` would suit this better, but that should be the last thing. Looks great overall!

Comment (onerandomusername) [reply docs/api.rst:783]: +1, don't forget to reorder it

Comment (ChristopherJHart) [reply docs/api.rst]: Done!

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**